docs/modules/ROOT/pages/how-tos/configuring/component-nudges.adoc

```diff
@@ -1 +1,81 @@
-= Defining component relationships
\ No newline at end of file
+= Defining component relationships
+
+When an application includes multiple components, their repositories might contain references to image builds of another component. For instance, an application might include multiple child components that, using a Dockerfile `FROM` clause, reference the image build of a common parent.
+
+In such instances, whenever you build a new image of the common parent component, you need to update the references to it. Specifically, you need to copy and paste the pullspec and image digest of that new image to your other source repositories. This process is tedious and error-prone.
+
+If you build an application with {ProductName} and make references between components with image digests (`sha256:...`), {ProductName} can automatically generate pull requests to update those references. To use this functionality, simply define the relationships between your components, either in the {ProductName} UI, or in your CLI.
```

Comment:
By default, it only looks in certain files for these references: https://github.com/konflux-ci/build-service/blob/0e591f628aab9cbe9fac2b9bdc8883e955a560e7/controllers/component_dependency_update_controller.go#L287C18-L287C60 ... which can be customized by setting an Annotation on the pipeline definition: https://github.com/konflux-ci/build-service/blob/0e591f628aab9cbe9fac2b9bdc8883e955a560e7/controllers/component_dependency_update_controller.go#L65

Reply:
I think I follow what you're saying. For example, users could delete '.*.yaml' if, for some reason, they don't want Konflux to search for references in yaml files, right? Or add another file type to look for.

But how would users change that definition, given that it's in the build service? They only clone and use this repo: https://github.com/konflux-ci/konflux-ci
Is there a file in that repo with the same annotation to edit?

Reply:
The would just add an annotation to their PipelineDefinition, i.e. in here: https://github.com/spork-madness/common-nodejs-parent/blob/main/.tekton/common-nodejs-parent-push.yaml#L4

This is more likely to be used if you want to nudge references in different files.

Reply:
I took a stab at covering this in a note. Please take a look and see what you think.